Base branch for refactoring

saledjenic opened the pull request:
All refactor branches should be branch of this one.

PR: `status-go` [here](https://github.com/status-im/status-go/pull/2423)

saledjenic committed:
added new backend for refactoring purpose, contacts added

saledjenic committed:
initial chat class added

saledjenic committed:
initial community class added

michaelsbradleyjr commented on status/statusgo_backend_new/contacts.nim:
```diff
@@ -0,0 +1,41 @@
+  result = callPrivateRPC("contacts".prefix, payload)
+
+proc getContactById*(id: string): RpcResponse[JsonNode] {.raises: [Exception].} =
+  let payload = %* []
```
Should `id` be sent in the payload?

```suggestion
  let payload = %* [id]
```

saledjenic replied:
@michaelsbradleyjr yes, you're absolutely right, most likely that is missed cause that method is just added, but not used yet.

saledjenic committed:
initial accounts class added

saledjenic committed:
login method added

alaibe committed:
Refactor/wallet 1 (#89)

* refactor: add custom token new backend

* refactor: add transactions new backend

* refactor: add collectible new backend

* refactor: add accounts backend

* refactor: add settings backend

* refactor: Add eth call to fetch balance

* refactor: add call to get eth block

richard-ramos committed:
refactor: remove bookmarks (#90)

richard-ramos committed:
refactor: dapp permissions (#92)

alaibe committed:
Refactor/wallet part 2 (#91)

* refactor: add save account

* refactor: add account generation

* refactor: save settings

* refactor: add update account in new be

* add getTransfersByAddress (#93)

Co-authored-by: Jonathan Rainville <rainville.jonathan@gmail.com>

saledjenic committed:
initial messages class added

alaibe committed:
fix: transaction request loading contracts (#96)

saledjenic committed:
add/remove reactions added

saledjenic committed:
- pin/unpin message added
- fetch message's details by message id added
- fetch reactions for message with id added

iurimatias committed:
bump status-go
